solutions/leap_year_checker.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+A module for checking if a given year is a leap year.
+
+Module contents:
+    - is_leap_year: A function that checks if a given year is a leap year.
+
+Created on 2024-12-31
+Author: Hussaini Ahmed
+"""
+
+
+def is_leap_year(year: int) -> bool:
+    """
+    Checks if a given year is a leap year.
+
+    A leap year:
+    - Is divisible by 4, but not divisible by 100, unless it is also divisible by 400.
+
+    Parameters:
+        year (int): The year to check. Must be a positive integer.
+
+    Returns:
+        bool: True if the year is a leap year, False otherwise.
+
+    Raises:
+        TypeError: If the input is not an integer.
+        ValueError: If the year is not a positive integer.
+
+    Examples:
+        >>> is_leap_year(2020)
+        True
+        >>> is_leap_year(1900)
+        False
+        >>> is_leap_year(2000)
+        True
+        >>> is_leap_year(2023)
+        False
+        >>> is_leap_year(0)
+        False
+    """
+    if not isinstance(year, int):
+        raise TypeError("The year must be an integer.")
+    if year <= 0:
+        raise ValueError("The year must be a positive integer.")
```

Comment:
It would be better to omit using logic in assertions. Additionally, you can use 'assert'.
```suggestion
   assert isinstance(year, int), "The year must be an integer."
   assert year > 0, "The year must be a positive integer" 
```